processor/batchprocessor/factory.go

```diff
@@ -56,7 +57,7 @@ func createTracesProcessor(
 	nextConsumer consumer.Traces,
 ) (component.TracesProcessor, error) {
 	level := set.MetricsLevel
-	return newBatchTracesProcessor(set, nextConsumer, cfg.(*Config), level)
+	return newBatchTracesProcessor(set, nextConsumer, cfg.(*Config), level, featuregate.GetRegistry())
```

Comment:
why pass in the registry here instead of getting the registry within the new func?

Reply:
Mainly because tests uses these `newBatch*Processor` functions, so in order to change the featuregate registry during tests I had to update it. 

No strong feeling about keep this or changing tests to use the code inside the `newBatchProcessor`, I think I prefer the current definition.

Reply:
@codeboten this is a technique that we did in lots of other places. We pass the registry so that unit-tests can run in parallel and not have to deal with cleanups.

Reply:
sounds good.